includes/class-wp-job-manager-shortcodes.php

```diff
@@ -272,6 +274,16 @@ public function job_dashboard( $atts ) {
 		return ob_get_clean();
 	}
 
+	/**
+	 * Filters the url from paginate_links to avoid multiple calls for same action in job dashboard
+	 *
+	 * @param string $link
+	 * @return string
+	 */
+	public function filter_paginate_links( $link ) {
+		return remove_query_arg( [ 'action', 'job_id', '_wpnonce' ], $link );
+	}
```

Comment:
I find myself concerned with this again. I had imagined before that this would only be called if a WPJM-related shortcode existed on the page, but looking again I think this would be called on _any_ page that calls the `paginate_links` function. This seems dangerous to me.

Do you think it would make sense to add a condition here to ensure that it only happens on the page(s) we want? E.g. see the `shortcode_action_handler` method. Maybe something like:

```suggestion
	public function filter_paginate_links( $link ) {
		global $post;
		
		if ( is_page() && has_shortcode( $post->post_content, 'job_dashboard' ) ) {
			return remove_query_arg( [ 'action', 'job_id', '_wpnonce' ], $link );
		}
		
		return $link;
	}
```

If so, maybe we could also extract that conditional logic to a private method (assuming we use the same logic as in `shortcode_action_handler`). Maybe `private function is_job_dashboard_page() { ... }`

Reply:
Hey Alex! 
You're totally right! It's a bit too aggressive... Your suggestion works perfectly! I just implemented it...